openspec/changes/add-model-provider-cred-store-oagw/specs/cred-store/spec.md

```diff
@@ -0,0 +1,843 @@
+  - `created_at timestamptz not null`
+  - `updated_at timestamptz not null`
+
+The Vault-backed plugin SHALL enforce access control by scoping the reference table queries (tenant/user) using secure ORM.
+
+#### Scenario: Vault-backed plugin write
+- **WHEN** the gateway calls `CredStorePluginApi::upsert_secret(ctx, secret_id, secret_type_id, secret_ciphertext, parameters)`
+- **THEN** the plugin scopes the operation to the tenant/user in `SecurityCtx`
+- **AND** the plugin writes the secret to Vault under a tenant/user scoped path
+- **AND** the plugin stores only a Vault reference in `cred_store_vault_secret_ref`
+
+#### Scenario: Vault-backed plugin read
+- **WHEN** the gateway calls `CredStorePluginApi::get_secret_material(ctx, secret_id, secret_type_id)`
+- **THEN** the plugin resolves the Vault reference under secure ORM scope
+- **AND** the plugin reads/decrypts the secret material from Vault
+
+### Requirement: global and per-tenant configuration
+
+All the parameters, such as timeouts, retries, etc must be configurable globally (as serviec config) and on per-tenant basis.
+
+### Requirement: cred_store audit logging
+The system SHALL maintain an immutable audit log for all secret operations.
+
+The system SHALL persist audit records in a separate append-only table.
+
+#### Database table: cred_store_audit_log
+- `id uuid pk`
+- `tenant_id uuid not null`
+- `client_id uuid null`
+- `is_user_operation bool not null` (true for user-initiated operations, false for system-initiated operations)
+- `operation text not null` (enum: 'create', 'read', 'update', 'delete', 'rotate')
+- `secret_id uuid not null`
+- `secret_type_id uuid not null`
+- `status text not null` (enum: 'success', 'failure')
+- `error_code text null`
+- `source_ip inet null`
+- `user_agent text null`
+- `trace_id text null`
+- `context_metadata jsonb null`
+- `timestamp timestamptz not null default now()`
+
+**Indexes:**
+- `(tenant_id, timestamp desc)` for tenant audit queries
+- `(secret_id, timestamp desc)` for secret-specific audit trail
+- `(client_id, timestamp desc)` for user activity queries
+
+NOTE: table parititioning and archiving and retention policies are not specified in this document. It's a subject for future improvements.
+
+#### Scenario: Audit log write on secret read
+- **WHEN** user/client U reads secret S via `get_secret()`
+- **THEN** the gateway inserts a row into `cred_store_audit_log`
+- **AND** captures (tenant_id, client_id, 'read', secret_id, secret_type_id, 'success', timestamp)
+- **AND** includes OpenTelemetry trace_id from current span
+
+#### Scenario: Audit log for failed access
+- **WHEN** user U attempts to read secret S but is denied (403)
+- **THEN** the gateway logs (client_id, 'read', secret_id, 'failure', 'access_denied')
+
+### Requirement: cred_store audit retention
+The system SHOULD support configurable audit log retention policies.
+
+Retention SHALL be configurable per tenant with defaults:
+- Default retention: 90 days
```

Comment:
Let's define a single policy model with explicit precedence: global default, per-tenant override, per-secret-type override and etc. Right now this reads like two unrelated rules.